app/models/user.rb

```diff
@@ -1,5 +1,9 @@
 class User < ApplicationRecord
   has_many :posts, foreign_key: :author_id
-  has_many :comments, foreign_key: :user_id
-  has_many :likes, foreign_key: :user_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+  has_many :posts, through: :likes, foreign_key: :author_id
+  has_many :posts, through: :comments, foreign_key: :author_id
```

Comment:
- [ ] These 2 lines are not necessary for the associations to work, this is only needed with many to many relationships which this is not.